src/journal/journald-file.c

```diff
@@ -276,19 +276,24 @@ int journald_file_rotate(
                 bool seal,
                 Set *deferred_closes) {
 
+        _cleanup_free_ char *path = NULL;
         JournaldFile *new_file = NULL;
         int r;
 
         assert(f);
         assert(*f);
 
+        path = strdup((*f)->file->path);
+        if (!path)
+                return -ENOMEM;
```

Comment:
This is a bit ugly because we do a totally pointless strdup() here. We don't rotate files often enough for this to matter at all, but it still seems a bit sloppy. Maybe change `journal_file_archive()` to take an additional output pointer where it will donate the old path pointer?